src/openloco/windows/townwnd.cpp

```diff
@@ -1,31 +1,791 @@
         }
-#endif
     }
 
     // 0x00499B7E
-    // dx: townId
-    // esi: {return}
-    window* open_town_window(uint16_t townId)
+    window* open(uint16_t townId)
+    {
+        auto window = WindowManager::bringToFront(WindowType::town, townId);
+        if (window != nullptr)
+        {
+            if (input::is_tool_active(window->type, window->number))
+                input::cancel_tool();
+
+            window = WindowManager::bringToFront(WindowType::town, townId);
+        }
+
+        if (window == nullptr)
+        {
+            // 0x00499C0D start
+            window = WindowManager::createWindow(WindowType::town, windowSize, 0, &town::events);
+            window->widgets = town::widgets;
+            window->enabled_widgets = town::enabledWidgets;
+            window->number = townId;
+            window->current_tab = 0;
+            window->frame_no = 0;
+            window->disabled_widgets = 0;
+            window->min_width = 192;
+            window->min_height = 161;
+            window->max_width = 600;
+            window->max_height = 440;
+            window->flags |= window_flags::resizable;
+
+            auto skin = objectmgr::get<interface_skin_object>();
+            if (skin != nullptr)
+            {
+                window->colours[0] = skin->colour_0B;
+                window->colours[1] = skin->colour_0C;
+            }
+            // 0x00499C0D end
+
+            window->var_848 = -1;
+        }
+
+        // TODO(avgeffen): only needs to be called once.
+        common::initEvents();
+
+        window->current_tab = 0;
+        window->invalidate();
+
+        window->widgets = town::widgets;
+        window->enabled_widgets = town::enabledWidgets;
+        window->holdable_widgets = 0;
+        window->event_handlers = &town::events;
+        window->activated_widgets = 0;
+        window->disabled_widgets = 0;
+        window->init_scroll_widgets();
+        town::initViewport(window);
+
+        return window;
+    }
+
+    namespace population
     {
-        registers regs;
-        regs.dx = townId;
-        call(0x00499B7E, regs);
-        return (window*)regs.esi;
+        static widget_t widgets[] = {
+            commonWidgets(223, 161, string_ids::title_town_population),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        // 0x00499469
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+        }
+
+        // 0x004994F9
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            gfx::drawpixelinfo_t* clipped = nullptr;
+            if (!gfx::clip_drawpixelinfo(&clipped, dpi, self->x, self->y + 44, self->width, self->height - 44))
+                return;
+
+            auto town = townmgr::get(self->number);
+            *(int32_t*)&*commonFormatArgs = town->history_min_population;
```

Comment:
Would rather have this in its own variable, and set the formatarg for each label